### PR TITLE
#408

### PR DESCRIPTION
**Pull request by HDouss**
Resolving issue #408 
Implements graph node connections
Update the puzzle to continue with Java xml generation
Add a new puzzle to build connections based on the correct overloaded method.

### CODE REVIEW

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) everybody who has role `REV` is banned at [#410](https://github.com/yegor256/jpeek/issues/410); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by HDouss** [comment]
@paulodamaso how can we move on here? Could you assign yourself?

**Comment by paulodamaso** [comment]
@0crat assign me

File: src/main/java/org/jpeek/graph/XmlGraph.java [hunk 1]
```diff
@@ -66,16 +68,39 @@ public XmlGraph(final Skeleton skeleton) throws IOException {
...
      */
     private static List<Node> build(final Skeleton skeleton) throws IOException {
-        return new Mapped<XML, Node>(
+        final String clas = skeleton.xml().xpath("//class/@id").get(0);
+        final Map<XML, Node> byxml = new org.cactoos.map.Sticky<>(
+            method -> method,
             method -> new Node.Simple(
                 new Joined(
-                    "", method.xpath("@name").get(0), method.xpath("@desc").get(0)
+                    "", clas, ".", method.xpath("@name").get(0)
```
Comment (paulodamaso) [inline src/main/java/org/jpeek/graph/XmlGraph.java]: @HDouss You could inline `clas` variable here

**Comment by 0crat** [comment]
> @0crat assign me ([here](https://github.com/yegor256/jpeek/issues/410#issuecomment-595531451))

@paulodamaso This pull request [#410](https://github.com/yegor256/jpeek/issues/410) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/24a57ece-590e-4528-bcef-feab03f77aea); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/org/jpeek/graph/XmlGraph.java (hunk 1 above)
Comment (HDouss) [reply src/main/java/org/jpeek/graph/XmlGraph.java]: @paulodamaso Thanks! Fixed.

**Comment by paulodamaso** [comment]
@HDouss Great, thanks!

**Comment by 0crat** [comment]
Job was finished in 11 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Pull request merged**